src/healthkit.cs

```diff
@@ -3797,4 +4185,243 @@ interface HKVerifiableClinicalRecordSubject : NSSecureCoding, NSCopying
 		NSDateComponents DateOfBirthComponents { get; }
 	}
 
+	[Watch (9,0), MacCatalyst (16,0), Mac (13,0), iOS (16,0), NoTV]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface HKAttachment : NSSecureCoding, NSCopying
+	{
+		[Export ("identifier", ArgumentSemantic.Copy)]
+		NSUuid Identifier { get; }
+
+		[Export ("name")]
+		string Name { get; }
+
+		[Export ("contentType", ArgumentSemantic.Copy)]
+		UTType ContentType { get; }
+
+		[Export ("size")]
+		nint Size { get; }
+
+		[Export ("creationDate", ArgumentSemantic.Copy)]
+		NSDate CreationDate { get; }
+
+		[NullAllowed, Export ("metadata", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSObject> Metadata { get; }
+	}
+
+	[Watch (9,0), MacCatalyst (16,0), Mac (13,0), iOS (16,0), NoTV]
+	[BaseType (typeof (NSObject))]
+	interface HKAttachmentStore
+	{
+		[Export ("initWithHealthStore:")]
+		NativeHandle Constructor (HKHealthStore healthStore);
+
+		[Async]
+		[Export ("addAttachmentToObject:name:contentType:URL:metadata:completion:")]
+		void AddAttachment (HKObject @object, string name, UTType contentType, NSUrl URL, [NullAllowed] NSDictionary<NSString, NSObject> metadata, Action<HKAttachment, NSError> completion);
+
+		[Async]
+		[Export ("removeAttachment:fromObject:completion:")]
+		void RemoveAttachment (HKAttachment attachment, HKObject @object, Action<bool, NSError> completion);
```

Comment:
what does the bool represents here? Probably this is a good candidate to do a delegate instead of an Action, same for the rest in this class.